torchvision/transforms/functional.py

```diff
@@ -19,6 +21,15 @@
 from . import functional_tensor as F_t
 
 
+def log_api_usage_once(fn: Callable) -> Callable:
```

Comment:
shall we move this decorator to utils? or turn `_log_api_usage_once` into a decorator if it works for class as well?

Reply:
@pmeier I believe if you add this on the functional side, it will break JIT-scriptability. At least that's what I recall from Kai's earlier experiments (see #4976 and #5052). But then the CI tests are not currently failing which is very strange.

@kazhang any thoughts on whether the approach followed by Philip actually works OR if we have a gap on our testing strategy?

Reply:
Now I remember why decorator didn't work for me in #4976
See [my comment](https://github.com/pytorch/vision/pull/4976#issuecomment-976039784):
> Unfortunately, torchscript doesn't like decorator. When using the decorator on boxes functions, JIT looks for helper functions definition(e.g. _upcast) in utils.py


In this PR, we define the decorator in `functional.py` instead of `utils.py`, so JIT is still able to find helper functions used by the function to be scripted. So basically we'd need to define this decorator everywhere, not too bad compared with current solution....

Reply:
In #5436 I tried to move the decorator to utils.py, and reproduced the error: https://app.circleci.com/pipelines/github/pytorch/vision/14851/workflows/ad9beb44-ba9e-472e-a71b-8775e909f8b5/jobs/1195207

Reply:
Looking through the issue tracker, I found the long standing issue pytorch/pytorch#38964 that should have been fixed in pytorch/pytorch#63248. Maybe @qihqi can have a look if the fix was incomplete or our setup is different?